core/dbt/context/base.py

```diff
@@ -556,6 +557,15 @@ def flags(self) -> Any:
         """
         return flags
 
+    @contextproperty
+    def selected_resources(self) -> Any:
```

Comment:
I'm pretty sure that the base context is the wrong context for this functionality. We don't need this in yaml rendering and other non-sql rendering. It probably ought to go in a provider context. Is there some reason that you put it in base?

Reply:
I have just moved it to the ProviderContext